src/modules/iKinGazeCtrl/src/main.cpp

```diff
@@ -1142,7 +1142,7 @@ class GazeModule: public RFModule
         commData.robotName=rf.check("robot",Value("icub")).asString();
         commData.eyeTiltLim[0]=eyeTiltGroup.check("min",Value(-20.0)).asFloat64();
         commData.eyeTiltLim[1]=eyeTiltGroup.check("max",Value(15.0)).asFloat64();
-        commData.head_version=constrainHeadVersion(rf.check("head_version",Value(1.0)).asFloat64());
+        commData.head_version=constrainHeadVersion(iKinLimbVersion(rf.check("head_version",Value("1.0")).asString()));
```

Comment:
This is potentially breaking.
However, the input param `1.0` from the command line should be interpreted as a string without problems.

Reply:
Unfortunately, `--head_version 1.0` or `--head_version "1.0"` seem not to work 😔
I need more testing. A possibility is to rely on `--head_version v1.0` but is a breaking change, as pretty much as "1.0".

Reply:
The following simple snippet demonstrates the use case:
```c++
#include <cstdlib>
#include <yarp/os/ResourceFinder.h>
#include <yarp/os/LogStream.h>

int main(int argc, char* argv[]) {
    yarp::os::ResourceFinder rf;
    rf.configure(argc, argv);
    auto option = rf.check("option", yarp::os::Value("1.0")).asString();
    yInfo() << "option:" << option;
    return EXIT_SUCCESS;
}
```

Outputs:
```console
$ test_2
[INFO] option: 1.0
$ test_2 --option 2.0
[INFO] option: 
$ test_2 --option "2.0"
[INFO] option: 
$ test_2 --option "v2.0"
[INFO] option: v2.0
```

Reply:
My understanding is that if I enforce `.asString()`, everything passed in ought to be interpreted as a string, actually, even in the format w/o quotes.

cc @Nicogene

Reply:
>My understanding is that if I enforce .asString(), everything passed in ought to be interpreted as a string, actually, even in the format w/o quotes.

Probably not because `.asString()` is only a way to _re-interpret_ the content of the Value, which was already filled in.

@Nicogene do you know if there exists a simple way to get around it?

Reply:
What about getting the version with `rf.check("head_version",Value(1.0)).asFloat64())` as before, and then convert it to string with [`std::to_string`](https://en.cppreference.com/w/cpp/string/basic_string/to_string) for passing it to `iKinLimbVersion`?

Reply:
I would also add this version check tests in the icub-main unit tests

Reply:
This won't work as 2.1==2.10 for floats, neglecting the precision.

I thought about extending the tests but it won't be worthwhile covering an extremely little part of iKin.

The change will be breaking, most likely. No problem, I'll handle that.